File: hw/rojas/u2/hw19Inheritance/hw19InheritancePythonAndJavaScript/hw20Python/ec.edu.espe.view/FarmSystem.py
"""
    Author: Josue Rojas
    Project: Farm Simulator
"""
import sys
import os
import json
from datetime import datetime

current_dir = os.path.dirname(os.path.abspath(__file__))
model_path = os.path.join(current_dir, '../ec.edu.espe.model')
sys.path.append(model_path)

from Location import Location
from Cage import Cage
from Chicken import Chicken
from Cow import Cow
from Pig import Pig
from Sheep import Sheep

def load_animals_from_json(file_path, cage):
    animals = []
    
    try:
        with open(file_path, 'r') as file:
            data = json.load(file)
            
        for item in data:
            born_on = datetime.strptime(item['bornOn'], "%Y-%m-%d")
            
            if item['type'] == "Chicken":
                animal = Chicken(item['id'], item['breed'], born_on, item['gender'], 
                                 item['isAbleToReproduce'], item['weight'], cage, 
                                 item['isMolting'], item['layedEggs'])
                                 
            elif item['type'] == "Cow":
                animal = Cow(item['id'], item['breed'], born_on, item['gender'], 
                             item['isAbleToReproduce'], item['weight'], cage, 
                             item['isProducingMilk'], item['littersADay'])
                             
            elif item['type'] == "Pig":
                animal = Pig(item['id'], item['breed'], born_on, item['gender'], 
                             item['isAbleToReproduce'], item['weight'], cage)
                             
            elif item['type'] == "Sheep":
                last_shearing = datetime.strptime(item['lastShearing'], "%Y-%m-%d")
                animal = Sheep(item['id'], item['breed'], born_on, item['gender'], 
                               item['isAbleToReproduce'], item['weight'], cage, last_shearing)
            
            animals.append(animal)
            
    except FileNotFoundError:
        print(f"Error: File not found at {file_path}")
    except Exception as e:
        print(f"Error reading JSON: {e}")

    return animals

def main():
    print("\n==========================================")
    print("       F A R M   S I M U L A T O R        ")
    print("==========================================\n")
    loc = Location(10, 20)
    cage = Cage(1, loc)

    json_path = os.path.join(current_dir, 'animals.json')
    animals = load_animals_from_json(json_path, cage)

    if animals:
        for animal in animals:
            print(animal)
            print("-" * 40)

if __name__ == "__main__":
    main()

File: hw/rojas/u2/hw19Inheritance/Farmsystem/src/main/java/ec/edu/espe/farmsystem/view/FarmSystem.java
package ec.edu.espe.farmsystem.view;

import ec.edu.espe.farmsystem.model.Cage;
import ec.edu.espe.farmsystem.model.Chicken;
import ec.edu.espe.farmsystem.model.Cow;
import ec.edu.espe.farmsystem.model.FarmAnimal;
import ec.edu.espe.farmsystem.model.Location;
import ec.edu.espe.farmsystem.model.Pig;
import ec.edu.espe.farmsystem.model.Sheep;
import java.util.Date;

/**
 *
 * @author Josue Rojas
 */
public class FarmSystem {
    public static void main(String[] args) {
        int id;
        String breed;
        Date bornOn;
        boolean isAbleToReproduce;
        float weight;
        String gender;
        Cage cage;
        Location location;
        int xCoordinate;
        int yCoordinate;
        Date lastShearing;
        lastShearing = new Date(2006,11,4);
        xCoordinate = 10;
        yCoordinate = 20;
        isAbleToReproduce=false;
        weight = 10.4F;
        gender="male";
        bornOn = new Date(2025,2,1);
        breed= "Holstein";
        location = new Location(xCoordinate, yCoordinate);
        cage = new Cage(1,"Corral for cows", 2);
        
            FarmAnimal farmAnimal;
            farmAnimal = new Chicken(true, 0, 1, breed, bornOn, gender, isAbleToReproduce, weight, cage);
            System.out.println("farmAnimal --> " + farmAnimal);
            farmAnimal = new Cow(true, 1.5F, 2, breed, bornOn, gender, isAbleToReproduce, weight, cage);
            System.out.println("farmAnimal --> " + farmAnimal);
            farmAnimal = new Pig(3, breed, bornOn, gender, isAbleToReproduce, weight, cage);
            System.out.println("farmAnimal --> " + farmAnimal);
            farmAnimal= new Sheep(lastShearing,4, breed, bornOn, gender, isAbleToReproduce, weight, cage);
            System.out.println("farmAnimal --> " + farmAnimal);
        
        
    }
}